js/interactive-guides/utils.js

```diff
@@ -40,69 +40,94 @@ var utils = (function() {
-        //console.log("AAA string ", strAction);
-        var name;  
-        if (strAction.indexOf("name=") !== -1) {
-          var index = strAction.indexOf("name=") + 5;
-          //console.log("index ", index);
-          name = strAction.substring(index);
-          //console.log("name ", name);
-          var quote = name.substring(0, 1);
-          //console.log("quote ", quote);
-          var tmpString = name.substring(1);
-          //console.log("tmpString ", tmpString);
-          var lastIndex = tmpString.indexOf(quote);
-          name = name.substring(0, lastIndex + 2);
-          //console.log("name=", name);
+    var __getTitleAction = function(strAction) {
+        var title;  
+        if (strAction.indexOf("title=") !== -1) {
+            var index = strAction.indexOf("title=") + 6;
+            //console.log("index ", index);
+            title = strAction.substring(index);
+            var nextSpace = title.indexOf(" ");
+            // get content of title
+            if (nextSpace !== -1) {
+                title = title.substring(0, nextSpace);
+            }
+            // check if contain quote
+            if (title.indexOf("\"") !== -1 ||
+                title.indexOf("'") !== -1) {
+                //var quote = title.substring(0, 1);
+                //var tmpString = title.substring(1);
+                //console.log("tmpString ", tmpString);
+                //var lastIndex = tmpString.indexOf(quote);
+                //title = title.substring(0, lastIndex + 2);
+            } else { 
+                title = "'" + title + "'";
+            }
         }
-        return name;
+        //console.log("title=", title);
+        return title;
     };
     
     var __getCallbackAction = function(strAction) {
-        //console.log("BBB string ", strAction);
-        var callbackStr;
-        if (strAction.indexOf("callback=") !== -1) {
-            var index = strAction.indexOf("callback=") + 9;
-            //console.log("index ", index);
+        var callbackStr, index;
+           
+        if (strAction.indexOf("onclick=") !== -1) {
+            index = strAction.indexOf("onclick=") + 8;
+        } else if (strAction.indexOf("onkeypress=") !== -1) {
+            index = strAction.indexOf("onkeypress=") + 11;
+        }
+        //console.log("index ", index);
+        if (index) {
             callbackStr = strAction.substring(index);
             //console.log("callback ", callbackStr);
             var lastIndex = callbackStr.indexOf(")");
-            callbackStr = callbackStr.substring(0, lastIndex + 2);
-            //console.log("callback=", callbackStr);
+            var hasQuotes = callbackStr.substring(0, 1);
+            if (hasQuotes.indexOf("\"") !== -1 ||
+                hasQuotes.indexOf("'") !== -1) {
+                callbackStr = callbackStr.substring(0, lastIndex + 2);
+            } else {
+                callbackStr = "\"" + callbackStr.substring(0, lastIndex + 1) + "\"";
+            }
```

Comment:
Again the logic should be looking for opening single/double quote and then the corresponding ending quote.